app/gui2/src/components/GraphEditor/GraphVisualization.vue

```diff
@@ -56,20 +67,29 @@ function switchToDefaultPreprocessor() {
   visPreprocessor.value = DEFAULT_VISUALIZATION_CONFIGURATION
 }
 
+watch(
+  () => props.currentType,
+  () => (error.value = undefined),
+)
+
 watchEffect(async () => {
   if (props.currentType == null) {
     return
   }
 
   visualization.value = undefined
-  const module = await visualizationStore.get(props.currentType)
-  if (module) {
-    if (module.defaultPreprocessor != null) {
-      updatePreprocessor(...module.defaultPreprocessor)
-    } else {
-      switchToDefaultPreprocessor()
+  try {
+    const module = await visualizationStore.get(props.currentType).value
+    if (module) {
+      if (module.defaultPreprocessor != null) {
+        updatePreprocessor(...module.defaultPreprocessor)
+      } else {
+        switchToDefaultPreprocessor()
+      }
+      visualization.value = module.default
     }
-    visualization.value = module.default
+  } catch (error: any) {
+    error.value = error as Error
```

Comment:
an `instanceof Error` check would be better than a cast. Unexpected error types should simply be rethrown.